src/coreclr/jit/flowgraph.cpp

```diff
@@ -2367,8 +2367,13 @@ class MergedReturns
                 assert(searchLimit < maxReturns);
                 mergedReturnBlock = CreateReturnBB(searchLimit);
                 comp->genReturnBB = mergedReturnBlock;
-                // Downstream code expects the `genReturnBB` to always remain
-                // once created, so that it can redirect flow edges to it.
+
+                // In principle we should be able to remove the proctection
+                // we add here, and remove the genReturnBB if it becomes unreachable,
+                // since we now introduce all the flow to it during this phase.
```

Comment:
There's also some somewhat random code in lowering and codegen that references this block, so the deleted comment is still accurate.

Reply:
I think all those can be replaced by checks for the "right" BBJ_RETURN (the one and only without `BBF_HAS_JMP`), though I didn't want to push things quite that far.